src/core/PassphraseGenerator.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ *  Copyright (C) 2013 Felix Geyer <debfx@fobos.de>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PassphraseGenerator.h"
+
+#include <math.h>
+#include <QFile>
+#include <QTextStream>
+
+#include "crypto/Random.h"
+#include "core/FilePath.h"
+
+PassphraseGenerator::PassphraseGenerator()
+    : m_length(0)
+    , m_separator(' ')
+{
+    const QString path = filePath()->dataPath("wordlists/eff_large.wordlist");
+    setWordlist(path);
+}
+
+double PassphraseGenerator::calculateEntropy(QString passphrase)
+{
+    Q_UNUSED(passphrase);
+
+    if (m_wordlist.size() == 0) {
+        return 0;
+    }
+
+    return log(m_wordlist.size()) / log(2.0) * m_length;
+}
+
+void PassphraseGenerator::setLength(int length)
+{
+    m_length = length;
+}
+
+void PassphraseGenerator::setWordlist(QString path)
+{
+    m_wordlist.clear();
+
+    QFile file(path);
+    if (!file.open(QIODevice::ReadOnly | QIODevice::Text)) {
+        qWarning("Couldn't load passphrase wordlist.");
+        return;
+    }
+
+    QTextStream in(&file);
+    while (!in.atEnd()) {
+        m_wordlist.append(in.readLine());
+    }
+
+    if (m_wordlist.size() < 1000) {
```

Comment:
@louib The limit is for preventing users use small wordlist. The entropy-meter will let the user use a phassphrase with 5 word from a 10-word wordlist but the user can simply ignore the "Password Quality: Bad" label and go with it.